### PR TITLE
271 mobile article category system

### PR DESCRIPTION
**Pull request by hugolopespinto**
Item category is now available in the items descriptions
image

image

### CODE REVIEW

File: Configurateur/front/android/app/build.gradle [hunk 1]
```diff
@@ -26,7 +26,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion flutter.compileSdkVersion
+    compileSdkVersion 34
```
Comment (invpuppet) [inline Configurateur/front/android/app/build.gradle:29]: do we need to keep the same version for the compiler ?

File: Server/routes/Mobile/containers.js [hunk 2]
```diff
@@ -43,6 +43,7 @@ router.get('/:containerId/articleslist/', async (req, res) => {
     } else if (!container.items || container.items.length === 0) {
       return res.status(204).json({ message: 'Container doesn\'t have items' })
     }
+    console.log(container.items)
```
Comment (invpuppet) [inline Server/routes/Mobile/containers.js:46]: console.log

File: Server/tests/web/itemCategory.test.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+  it("Should create an item category", async () => {
+    jwtMiddleware.verifyToken.mockResolvedValueOnce();
+
+    const response = await supertest(app)
+      .post("/")
+      .set("Authorization", "Bearer mockedAccessToken")
+      .send({ name: "Nouvelle Catégorie" });
+
+    console.log(response.body);
```
Comment (invpuppet) [inline Server/tests/web/itemCategory.test.js:29]: console.log

File: Server/tests/web/itemCategory.test.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      .send(requestBody);
+
+    expect(response.status).toBe(400);
+    expect(jwtMiddleware.verifyToken).toHaveBeenCalledWith(
+      "Bearer mockedAccessToken",
+    );
+    expect(itemCategoryCtrl.deleteItemCategory).not.toHaveBeenCalled();
+  });
+
+});
```
Comment (invpuppet) [inline Server/tests/web/itemCategory.test.js:132]: new line

**Review by invpuppet — CHANGES_REQUESTED** [review]

File: Configurateur/front/android/app/build.gradle (hunk 1 above)
Comment (hugolopespinto) [reply Configurateur/front/android/app/build.gradle:29]: no,just here because the last version wasn't compiling

**Review by invpuppet — APPROVED** [review]

**Review by H-chauvet — APPROVED** [review]
Good

**Pull request merged**